src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -1951,13 +1951,34 @@ void Assembler::evcvtps2ph(Address dst, KRegister mask, XMMRegister src, int imm
   emit_int8(imm8);
 }
 
+void Assembler::vcvtps2ph(Address dst, XMMRegister src, int imm8, int vector_len) {
+  assert(VM_Version::supports_evex() || VM_Version::supports_f16c(), "");
+  InstructionMark im(this);
+  InstructionAttr attributes(vector_len, /* rex_w */ false, /* legacy_mode */ false, /* no_mask_reg */ true, /*uses_vl */ true);
+  attributes.set_address_attributes(/* tuple_type */ EVEX_HVM, /* input_size_in_bits */ EVEX_NObit);
```

Comment:
Is it correct to set `EVEX_*` attributes in case EVEX is switched off (by `UseAVX` flag)?

Reply:
Or a CPU supports F16C but does not EVEX (avx512f).

Reply:
Hi Vladimir, we have a prior example of vpaddb instruction where these attributes are set. The assembler will ignore these attributes if UseAVX < 3.

Reply:
Good. Thank you for answering my questions.